SPARCore/GADArticle.m

```diff
@@ -2,6 +2,97 @@
 
 @implementation GADArticle
 
+static NSString *const API_ARTICLE_ID = @"id";
+static NSString *const API_AUTHORS = @"authors";
+static NSString *const API_BRIEF = @"brief";
+static NSString *const API_CONTENT = @"content";
+static NSString *const API_DATE_PUBLISHED = @"datePublished";
+static NSString *const API_HEADER_IMAGE = @"headerImage";
+static NSString *const API_PUBLICATION_ID = @"publication";
+static NSString *const API_TITLE = @"title";
+
+static NSString *const API_PUBLICATION_PATH = @"publications";
+static NSString *const API_ARTICLE_PATH = @"articles";
+
++ (void) articlesFromPublication: (NSString *)publicationId
+               completionHandler:(void(^_Nonnull)(NSArray<GADArticle *>
+                                                  *_Nullable articles,
+                                                  NSError *_Nullable error))completion {
+    NSURL *queryURL = [self urlForArticlesFromPublication:publicationId];
+    
+    [super fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData) {
+                            return [self articlesFromJSON:jsonData];
+                         }
+                        completionHandler:completion];
+
+}
+
+- (void) fetchFulltextWithCompletion: (void(^_Nonnull)(GADRemoteModel * *_Nullable model,
+                                                NSError *_Nullable error))completion {
+    NSURL *queryURL = [GADArticle createURLWithArticle:self.articleId publication:self.publicationId];
+    
+    [GADRemoteModel fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData){
+                             return [GADArticle articlesFromJSON:jsonData];
+                         }
+                        completionHandler:^(NSArray <GADArticle *> *_Nullable model, NSError *_Nullable error){
+                            GADArticle *fullArticle = model[0];
+                            self.content = fullArticle.content;
+                            self.authors = fullArticle.authors;
+                            //completion(); --How should we call the completion handler here?
+                        }];
+    
+}
+
++ (NSArray <GADArticle *> *) articlesFromJSON: (NSData *)json {
```

Comment:
This sequence of methods (`articlesFromJSON:`, `articleFromDictionary:`, etc) are *excellent*.